### PR TITLE
722 clear all components

### PR DESCRIPTION
**Pull request by chirag729**
# Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change. If this is a breaking change, specify explicitly which APIs have been changed*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [ ] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flutter Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [ ] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

### CODE REVIEW

File: packages/flame/test/game/base_game_test.dart [hunk 1]
```diff
@@ -200,4 +200,20 @@ void main() {
...
+
+    game.onResize(size);
+    game.addAll(components);
+
+    // The components are not added to the component list until an update has been performed
+    game.update(0.0);
+    expect(game.components.length, equals(3));
+
+    game.clear();
+    game.update(0.0);
```
Comment (wolfenrain) [inline packages/flame/test/game/base_game_test.dart:216]: I would also add an `expect` before the update to ensure that the clear does not clear it directly.

```suggestion
    expect(game.components.length, equals(3));
    game.update(0.0);
```

**Review by wolfenrain — APPROVED** [review]
Missing a CHANGELOG entry, and one small nit.

File: packages/flame/test/game/base_game_test.dart (hunk 1 above)
Comment (chirag729) [reply packages/flame/test/game/base_game_test.dart:219]: Updates made as per your comments

**Review by erickzanardo — APPROVED** [review]

**Review by luanpotter — APPROVED** [review]
LGTM

**Pull request merged**